dmkitui/BucketList
# #148186169 Write Documentation, Implement feedback

#### What does this PR do?
Implement feedback given during project defence, complete README and api documentation.
#### Description of Task to be completed?
1. Complete the Readme.
2. Remove commented-out code from project code.
3. Eliminate pagination bug, where GET for a specified page was not working. Also add page information to the GET response when limit and pagination parameters are present in the request.
4. Improve tests coverage to 99%
5. Remove linter syntax highlights by complying with pep8 standards, especially white spaces.
6. Complete documentation using Slate.
7. Host the application on heroku: https://flask-api-bucketlist.herokuapp.com/ The homepage now hosts the API documentation.

#### How should this be manually tested?
1. Run the tests.
- Clone the repo, and prepare the environments as set out in the README.
- On the terminal, run `python manage.py test`
2. Visit the documentation [repo](https://github.com/dmkitui/slate/tree/develop/source) and follow instructions on how to get the slate server running.
3. Run POSTMAN, add several bucketlists, set parameter limit to 1 and page to 3. You should see the results as shown below.

#### Any background context you want to provide?
1. Coverage was initially at 95%
2. Slate documentation uses a server to display the api documentation. Currently working on how to host the same together with the flask server.

#### What are the relevant pivotal tracker stories?
#148186169
#### Screenshots (if appropriate)
<img width="793" alt="screen shot 2017-07-11 at 08 57 03" src="https://user-images.githubusercontent.com/28535137/28053231-094e5fc8-6618-11e7-9a29-cc0249cabda7.png">
<img width="754" alt="screen shot 2017-07-11 at 08 57 53" src="https://user-images.githubusercontent.com/28535137/28053232-094fc67e-6618-11e7-8ce9-9dc7968e3d10.png">
<img width="741" alt="screen shot 2017-07-11 at 08 58 11" src="https://user-images.githubusercontent.com/28535137/28053230-094b61e2-6618-11e7-881a-731273c871d9.png">
<img width="722" alt="screen shot 2017-07-11 at 08 58 26" src="https://user-images.githubusercontent.com/28535137/28053234-095121a4-6618-11e7-9241-c8590e4b6acf.png">
<img width="807" alt="screen shot 2017-07-11 at 09 04 31" src="https://user-images.githubusercontent.com/28535137/28053233-09510af2-6618-11e7-920b-ba65960052bc.png">
<img width="1440" alt="screen shot 2017-07-11 at 21 33 02" src="https://user-images.githubusercontent.com/28535137/28084401-2d982076-6681-11e7-8481-29f4cca03030.png">
